litani

```diff
@@ -260,6 +260,11 @@ def get_args():
             "help":
                 "how many times to retry loading the run in 1 second intervals"
                 " (Default: %(default)s)"
+        }, {
+            "flags": ["--out-file"],
+            "metavar": "F",
+            "type": pathlib.Path,
```

Comment:
Instead of `pathlib.Path`, I suggest you create a new helper function in this file called `_non_directory_path` which takes a single argument and does the validation you're doing here 🐳. Instead of `logging.error` and `exit`, it should just throw `ValueError` with the error message. If you try it yourself, you should see that it formats a usage/help message for you if you do it this way.

Reply:
(see the example [here](https://docs.python.org/3/library/argparse.html#type))

Reply:
Resolved